osu.Game/Rulesets/Mods/DifficultyBindable.cs

```diff
@@ -99,22 +99,31 @@ private void updateMaxValue()
             CurrentNumber.MaxValue = ExtendedLimits.Value && extendedMaxValue != null ? extendedMaxValue.Value : maxValue;
         }
 
-        public override void BindTo(Bindable<float?> them)
+        public override void CopyTo(Bindable<float?> them)
         {
             if (!(them is DifficultyBindable otherDifficultyBindable))
-                throw new InvalidOperationException($"Cannot bind to a non-{nameof(DifficultyBindable)}.");
+                throw new InvalidOperationException($"Cannot copy to a non-{nameof(DifficultyBindable)}.");
+
+            base.CopyTo(them);
 
-            ReadCurrentFromDifficulty = otherDifficultyBindable.ReadCurrentFromDifficulty;
+            otherDifficultyBindable.ReadCurrentFromDifficulty = ReadCurrentFromDifficulty;
 
             // the following max value copies are only safe as long as these values are effectively constants.
-            MaxValue = otherDifficultyBindable.maxValue;
-            ExtendedMaxValue = otherDifficultyBindable.extendedMaxValue;
+            otherDifficultyBindable.MaxValue = maxValue;
+            otherDifficultyBindable.ExtendedMaxValue = extendedMaxValue;
+        }
+
+        public override void BindTo(Bindable<float?> them)
+        {
+            if (!(them is DifficultyBindable otherDifficultyBindable))
+                throw new InvalidOperationException($"Cannot bind to a non-{nameof(DifficultyBindable)}.");
+
+            base.BindTo(them);
```

Comment:
This base call reordering is causing an actual test failure (`TestSceneModDifficultyAdjustSettings.TestResetToDefaults`). Was there a particular reason for this one?

Reply:
I was attempting to keep things in the same order they were happening before, but may have made a mistake.

Reply:
I guess the closest translation that 100% preserves the same order would be:

```diff
diff --git a/osu.Game/Rulesets/Mods/DifficultyBindable.cs b/osu.Game/Rulesets/Mods/DifficultyBindable.cs
index 2b3ba1dd13..1c5ddcafba 100644
--- a/osu.Game/Rulesets/Mods/DifficultyBindable.cs
+++ b/osu.Game/Rulesets/Mods/DifficultyBindable.cs
@@ -118,12 +118,12 @@ public override void BindTo(Bindable<float?> them)
             if (!(them is DifficultyBindable otherDifficultyBindable))
                 throw new InvalidOperationException($"Cannot bind to a non-{nameof(DifficultyBindable)}.");
 
-            base.BindTo(them);
-
+            them.CopyTo(this);
             ExtendedLimits.BindTarget = otherDifficultyBindable.ExtendedLimits;
 
             // the actual values need to be copied after the max value constraints.
             CurrentNumber.BindTarget = otherDifficultyBindable.CurrentNumber;
+            base.BindTo(them);
         }
 
         public override void UnbindFrom(IUnbindable them)
```

but it looks quite funny because the `base.BindTo()` will call `CopyTo()` again. Moving back the base call to the end doesn't seem to fail any test though, so maybe fine to just go back to the old ordering?

Reply:
Maybe I misunderstood here, but moving the `BindTo` alone does not seem to be enough. Your diff will work, but I'm wonder if that is better than what I had.

Reply:
Ok weird, an entirely different test seems to be failing now? I saw that failure locally when I was testing locally but presumed it to be unrelated, but apparently I was wrong. It's probably going to have to be the ugly solution, then.

Reply:
The test seems to fail regardless, will investigate further.

Reply:
I've fixed this failure.... https://github.com/peppy/osu/commit/f6e229c24dd1e730a88807b8edcb3b38997f68ca

I'm not sure how this ever worked though. There's a bindable in the constructor or the mod which updates the `MinValue`, and with the order of properties previously it was updating the minimum value before copying the new value in the tests.

Pretty haphazard setup there.

Reply:
The "fix" wasn't sitting well with me, so I dug further and found the actual framework-side regression. See https://github.com/ppy/osu-framework/pull/5542.

Pull linked above fixes the test failure in question for me without [f6e229c](https://github.com/ppy/osu/pull/21298/commits/f6e229c24dd1e730a88807b8edcb3b38997f68ca) applied.